pandas/_libs/groupby.pyx

```diff
@@ -825,27 +828,28 @@ def group_quantile(ndarray[float64_t, ndim=2] out,
     ngroups = len(out)
     counts = np.zeros(ngroups, dtype=np.int64)
     non_na_counts = np.zeros(ngroups, dtype=np.int64)
+    labels_for_lexsort = labels.copy()
+
+    # Put '-1' (NaN) labels as the last group so it does not interfere
+    # with the calculations.
+    if N > 0:
+        na_label_for_sorting = labels.max() + 1
 
     # First figure out the size of every group
     with nogil:
         for i in range(N):
             lab = labels[i]
-            if lab == -1:  # NA group label
+            if lab == -1:
+                labels_for_lexsort[i] = na_label_for_sorting
                 continue
 
             counts[lab] += 1
             if not mask[i]:
                 non_na_counts[lab] += 1
 
-    # Get an index of values sorted by labels and then values
-    if labels.any():
-        # Put '-1' (NaN) labels as the last group so it does not interfere
-        # with the calculations.
-        labels_for_lexsort = np.where(labels == -1, labels.max() + 1, labels)
-    else:
-        labels_for_lexsort = labels
```

Comment:
This conditional is strange - special casing the all 0 label case (which seems rare and specific) when we really care about -1 labels. Regardless, since we already check for -1 in loop above, no need to check with `any` and `where` here